apps/backend/src/modules/points-of-interest/points-of-interest.controller.ts

```diff
@@ -0,0 +1,274 @@
+import { APIPath, HTTPMethodEnum } from "~/libs/enums/enums.js";
+import {
+	type APIHandlerOptions,
+	type APIHandlerResponse,
+	BaseController,
+} from "~/libs/modules/controller/controller.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type Logger } from "~/libs/modules/logger/logger.js";
+import { type PointsOfInterestService } from "~/modules/points-of-interest/points-of-interest.service.js";
+
+import {
+	type PointsOfInterestRequestDto,
+	type PointsOfInterestResponseDto,
+} from "./libs/types/type.js";
+import {
+	pointOfInterestCreateValidationSchema,
+	pointOfInterestUpdateValidationSchema,
+} from "./libs/validation-schemas/validation-schemas.js";
+
+/**
+ * @swagger
+ * components:
+ *   schemas:
+ *     PointsOfInterest:
+ *       type: object
+ *       properties:
+ *         id:
+ *           type: number
+ *         name:
+ *           type: string
+ *           example: "Central Park"
+ *
+ */
+
+class PointsOfInterestController extends BaseController {
+	private pointsOfInterestService: PointsOfInterestService;
+
+	public constructor(
+		logger: Logger,
+		pointsOfInterestService: PointsOfInterestService,
+	) {
+		super(logger, APIPath.POINTS_OF_INTEREST);
+		this.pointsOfInterestService = pointsOfInterestService;
+
+		this.addRoute({
+			handler: this.create.bind(this),
+			method: HTTPMethodEnum.POST,
+			path: "/",
+			validation: {
+				body: pointOfInterestCreateValidationSchema,
+			},
+		});
+
+		this.addRoute({
+			handler: this.delete.bind(this),
+			method: HTTPMethodEnum.DELETE,
+			path: "/:id",
+		});
+
+		this.addRoute({
+			handler: this.find.bind(this),
+			method: HTTPMethodEnum.GET,
+			path: "/:id",
+		});
+
+		this.addRoute({
+			handler: this.findAll.bind(this),
+			method: HTTPMethodEnum.GET,
+			path: "/",
+		});
+
+		this.addRoute({
+			handler: this.update.bind(this),
```

Comment:
let's call it patch and the same for pointOfInterestPatchValidationSchema, not to be confused later
```suggestion
			handler: this.patch.bind(this),
```